### PR TITLE
Add "Windows Terminal here" to File Browser context menu

### PR DESCRIPTION
**Pull request by amb5l**
This PR provides a potential solution to issue [#10461](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/10461).

### CODE REVIEW

**Comment by alankilborn** [comment]
You didn't follow instructions in CONTRIBUTING.md.

Each PR must have an issue where proposed functionality is discussed.

**Comment by amb5l** [comment]
> You didn't follow instructions in CONTRIBUTING.md.
> 
> Each PR must have an issue where proposed functionality is discussed.

Apologies. This PR relates to issue [#10461](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/10461).

File: PowerEditor/src/WinControls/FileBrowser/fileBrowser.cpp [hunk 1]
```diff
@@ -337,6 +337,7 @@ void FileBrowser::initPopupMenus()
 	generic_string findInFile = pNativeSpeaker->getDlgLangMenuStr(FOLDERASWORKSPACE_NODE, nullptr, IDM_FILEBROWSER_FINDINFILES, FB_FINDINFILES);
 	generic_string explorerHere = pNativeSpeaker->getDlgLangMenuStr(FOLDERASWORKSPACE_NODE, nullptr, IDM_FILEBROWSER_EXPLORERHERE, FB_EXPLORERHERE);
 	generic_string cmdHere = pNativeSpeaker->getDlgLangMenuStr(FOLDERASWORKSPACE_NODE, nullptr, IDM_FILEBROWSER_CMDHERE, FB_CMDHERE);
+	generic_string wtHere = pNativeSpeaker->getDlgLangMenuStr(FOLDERASWORKSPACE_NODE, nullptr, IDM_FILEBROWSER_CMDHERE, FB_WTHERE);
```
Comment (alankilborn) [inline PowerEditor/src/WinControls/FileBrowser/fileBrowser.cpp:340]: Should be `WTHERE`, not `CMDHERE`

**Comment by alankilborn** [comment]
A _Run_ menu entry, bound to a keycombo for easy activation, would seem to suffice as an alternative:

image

**Comment by alankilborn** [comment]
Also, if you prefer PowerShell over Cmd, you can set that up as explained [HERE](https://npp-user-manual.org/docs/preferences/#:~:text=Changing%20the%20command%2Dline%20interpreter%20used) in the User Manual.

If you need BOTH PowerShell and Cmd, suggest the _Run_ menu approach for PS.

**Comment by alankilborn** [comment]
> not to the directory selected in the File Browser

What exactly is the "file browswer"?

**Comment by amb5l** [comment]
> NVM: You mean _Folder as Workspace_.

Apologies if this was confusing, the source seems to refer to it as "File Browser", but you're right.

I've developed a habit of using the Folder as Workspace as a work focused alternative to Windows' File Explorer, and launching Explorer and command prompts from in there is very useful to me.

**Comment by alankilborn** [comment]
It begs the question(s):

Should there be:

* Cmd
* PowerShell
* wt

Or should there be:

* Cmd
* PowerShell

Or should there be (only):

* wt

**Comment by amb5l** [comment]
> It begs the question(s):
> 
> Should there be:
> 
> * Cmd
> * PowerShell
> * wt
> 
> Or should there be:
> 
> * Cmd
> * PowerShell
> 
> Or should there be (only):
> 
> * wt

WT supports both Cmd and PowerShell. It should be good for WIndows 10/11 users. For older WIndows versions, a context menu option for PowerShell might be nice. I've implemented one and can push an update to this PR if that's OK?

In a perfect world I would have an XML file providing some control over the entries in the Folder As Workspace context menu. Coding that is probably outside of my comfort zone...

**Comment by amb5l** [comment]
> WT supports both Cmd and PowerShell. It should be good for WIndows 10/11 users.

WT should probably not be offered to pre-Windows 10 users. I am looking at this now.

**Comment by alankilborn** [comment]
>WT should probably not be offered to pre-Windows 10 users

How to say this...?
Pre-Win 10 does not need to be supported by Notepad++ currently.
If something happens to work on e.g. Win 7, then that's fine, but it isn't required to.
Design of new features should be geared to Win10/11.

**Comment by alankilborn** [comment]
>I've implemented one and can push an update to this PR if that's OK?

I've no official say-so here, so...don't ask me. :-)

General guidance: Put forward your best PR possible.  :-)

**Comment by alankilborn** [comment]
But... I don't know how well this fits in with the other areas where Notepad++ does "Command Prompt Here".  The way that works is that Cmd is officially supported and there are hacks to config files which allow that to be replaced with PowerShell.

**Comment by amb5l** [comment]
OK I've pushed an update that adds a PowerShell option (for Windows >= 7) and makes WT conditional on Windows >= 10.

**Comment by alankilborn** [comment]
>OK I've pushed an update that adds a PowerShell option (for Windows >= 7) and makes WT conditional on Windows >= 10.

Hmm, not sure that this is the correct path, due to what I said before https://github.com/notepad-plus-plus/notepad-plus-plus/pull/13926#issuecomment-1648270377 .

>Let's see what the maintainers say about this.

Indeed.

BTW, some of this discussion is good to have BEFORE the code change is even started -- potentially lessens misdirection and wasted/revised effort.

**Comment by amb5l** [comment]
> BTW, some of this discussion is good to have BEFORE the code change is even started -- potentially lessens misdirection and wasted/revised effort.

Indeed, I'm learning. I should have looked harder for an issue mentioning Windows Terminal, and started there.

On the other hand, my impatience has got me a personal fork of Notepad++ that has the feature I've wanted for the past year!

**Comment by donho** [comment]
CMD in "Folder As Workspace" show follow
```xml
    <GUIConfig name="commandLineInterpreter">X:\yourConsolePath\yourConsole.xex</GUIConfig>
```
Please see https://github.com/notepad-plus-plus/notepad-plus-plus/issues/10461#issuecomment-1652245958

**Comment by donho** [comment]
> CMD in "Folder As Workspace" show follow

New issue has been created: https://github.com/notepad-plus-plus/notepad-plus-plus/issues/13934

**Pull request closed without merging**